### ISSUE TITLE
UndeclaredVariableInspection quick fix places declarations in the wrong place

### ISSUE THREAD

**Issue by DevulskyA** [rubberduck-vba/Rubberduck#4216]
Error:Trying to declare WdKeydown  
```
'REMOVER
Public Sub AddKeyBinding()
    With Application
        CustomizationContext = ActiveDocument
        .KeyBindings.Add KeyCode:=BuildKeyCode(wdKeyControl, wdKeyShift,
**Dim WdKeydown As Variant**  ' <=== RUBBERDUCK fix
WdKeydown),_
        KeyCategory:=wdKeyCategoryCommand,Command:="FORMTOVAR"
    End With
End Sub
```
---
```
'REMOVER
Public Sub AddKeyBinding()
    With Application
        .KeyBindings.Add KeyCode:=BuildKeyCode(wdKeyControl, wdKeyShift, WdKeydown), _
         KeyCategory:=wdKeyCategoryCommand,Command:="FORMTOVAR"
    End With
End Sub
```
---
Rubberduck v2.2.0.3505-pre

**Comment by comintern** [comment]
What build are you running? This should have been already been fixed in RD next.

![image](https://user-images.githubusercontent.com/8944005/42976654-7281866e-8b88-11e8-96f3-93505f215865.png)

**Comment by comintern** [comment]
The quick-fix for the triggered `UndeclaredVariableInspection` **_is_** broken though. It shouldn't be adding a `Dim` inside another context.

This is a stripped down example without the nested functions, named parameters, and line continuations:

```
Public Sub Example()
    Debug.Print Bar(x, y, z)
End Sub

Public Sub TestSubjectAfterQuickFix()
    Debug.Print Bar(
Dim x As Variant
x,
Dim y As Variant
y,
Dim z As Variant
z)
End Sub

Private Function Bar(x As Integer, y As Integer, z As Integer) As Integer
    Bar = x + y + z
End Function
```

**Comment by DevulskyA** [comment]
I am using Rubberduck v2.2.0.3505-pre

**Comment by MDoerner** [comment]
The quickfix bug is exactly issue #4215.

**Comment by comintern** [comment]
Closed with  #4218.
